src/EurekaClient.js

```diff
@@ -303,19 +303,21 @@ export default class Eureka {
    */
   transformRegistry(registry) {
     if (!registry) {
-      throw new Error('Unable to transform empty registry');
-    }
-    this.cache = { app: {}, vip: {} };
-    if (!registry.applications.application) {
-      return;
-    }
-    if (registry.applications.application.length) {
-      for (let i = 0; i < registry.applications.application.length; i++) {
-        const app = registry.applications.application[i];
-        this.transformApp(app);
-      }
+      this.logger.warn('Unable to transform empty registry');
     } else {
-      this.transformApp(registry.applications.application);
+      const newCache = { app: {}, vip: {} };
+      if (!registry.applications.application) {
+        return;
+      }
+      if (registry.applications.application.length) {
+        for (let i = 0; i < registry.applications.application.length; i++) {
```

Comment:
Switched. We do still need the first length check because of the not-so-great JSON. `registry.applications.application` could be an object or an array.